experimental/Schemes/Misc.jl

```diff
@@ -0,0 +1,249 @@
+  # bisection in order to find the optimal power g^k for 
+  # which this happens for the first time
+
+  k = 2^(length(powers)-1)
+  upper = pop!(powers)
+  if length(powers) == 0
+    # In this case g = a * f with exponent 1. No need for 
+    # further bisection.
+    return (1, a)
+  end
+  l = 2^(length(powers)-1)
+  lower = pop!(powers)
+  b = one(R)
+  while length(powers) > 0
+    # println( "looking between the $l-th power and the $k-th power..." )
+    power = pop!(powers)
+    middle = lower*power
+    (check, b) = divides( middle, f )
+    if check == true
+      upper = middle
+      a = b
+      k -= 2^length(powers)
+    else
+      lower = middle
+      l += 2^length(powers)
+    end
+  end
+  return (k, a)
+end
+
+# TODO: Up to now, this is merely a copy of the above routine. 
+function divides_power( f::T, g::T, check_radical_membership::Bool=false ) where{ T<:MPolyQuoElem }
+  # println( "Check whether some power of $g is divisible by $f" )
+  R = parent(f)
+  parent(f) == parent(g) || error( "elements are not contained in the same ring" )
+
+  # Test for radical membership in the first place.
+  !check_radical_membership || ( radical_membership( lift(g), modulus(parent(f)) + ideal( parent(lift(f)), lift(f) )) || return (-1, zero(R)))
+  # println( "Test for radical membership was positive." )
+
+  # Check successivly higher powers of g for containment in ⟨f⟩. 
+  # Once such a power is found, try to decrease it again to 
+  # find the optimum.
+
+  # the first dummy check that can not be well fit in the inductive 
+  # routine below:
+  check = false
+  a = zero(R)
+  check, a = divides( one(parent(f)), f )
+  if check
+    return (0,a)
+  end
+
+  powers = [g]
+  while true
+    # println( "looking at the $(2^(length(powers)-1))-th power..." )
+    global crash = ( last(powers), f )
+    check, a = divides( last(powers), f )
+    if check
+      break
+    end
+    push!( powers, last(powers)^2 )
+  end
+
+  # Now a power g^(2^(l-1)) is found that is divided by f with 
+  # l being the length of the vector powers. We use logarithmic 
+  # bisection in order to find the optimal power g^k for 
+  # which this happens for the first time
+
+  k = 2^(length(powers)-1)
+  upper = pop!(powers)
+  if length(powers) == 0
+    # In this case g = a * f with exponent 1. No need for 
+    # further bisection.
+    return (1, a)
+  end
+  l = 2^(length(powers)-1)
+  lower = pop!(powers)
+  b = one(R)
+  while length(powers) > 0
+    # println( "looking between the $l-th power and the $k-th power..." )
+    power = pop!(powers)
+    middle = lower*power
+    (check, b) = divides( middle, f )
+    if check == true
+      upper = middle
+      a = b
+      k -= 2^length(powers)
+    else
+      lower = middle
+      l += 2^length(powers)
+    end
+  end
+  return (k, a)
+end
+
+
+@Markdown.doc """
+    function diag( d::Vector{MatSpaceElem} )
+
+Writes the matrices in the list d as blocks on the 'diagonal' of 
+a new matrix D and returns D. 
```

Comment:
```suggestion
Writes the matrices in the list `d` as blocks on the 'diagonal' of 
a new matrix `D` and returns `D`. 
```
Why is "diagonal" quoted?